src/thread_task.rs

```diff
@@ -443,9 +442,14 @@ pub(crate) fn rav1d_task_delayed_fg(
         delayed_fg.in_0 = in_0.clone();
         delayed_fg.out = out.clone();
         delayed_fg.type_0 = TaskType::FgPrep;
-        delayed_fg.grain = match out.p.bpc {
-            8 => Grain::Bpc8(Default::default()),
-            10 | 12 => Grain::Bpc16(Default::default()),
+
+        // This initialization is done once per call to [`dav1d_apply_grain`].
+        // Nevertheless, it is a 48 KB zero initialization that C avoids,
+        // so we avoid reinitializing here if it already has the correct variant.
+        match (out.p.bpc, &delayed_fg.grain) {
```

Comment:
```suggestion
        #[cfg(all(feature = "bitdepth_8", feature = "bitdepth_16"))]
        match (out.p.bpc, &delayed_fg.grain) {
```

Reply:
Ah yeah. Added that along with a note about why we `cfg` out the entire match when only one bitdepth is enabled, since that wasn't obvious to me initially.